### ISSUE TITLE
Modifying the Subject Distinguished Name (DN) partially not working

### ISSUE THREAD

**Issue by Thommii** [Sleepw4lker/TameMyCerts#7]
Hello,
I have successfully installed and set up TameMyCerts (Version 1.3.683.747) (test environment).
Roughly everything works, but modifying the Subject Distinguished Name (DN) partially not working...

I created the following policy for testing the certificate template:
```
<!--
  This is an example configuration file for computer online certificate requests.
  Ensures that all certificates issued for this template expire on 31st December 2022, 23:59:59 (Europe/Berlin time).
  Works the same way for offline requests, but should include syntax and directory mapping rules there.
-->
<CertificateRequestPolicy xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema">
  <DirectoryServicesMapping>
    <SubjectDistinguishedName>
        <RelativeDistinguishedName>
            <Field>commonName</Field>
            <DirectoryServicesAttribute>dNSHostName</DirectoryServicesAttribute>
            <Mandatory>true</Mandatory>
        </RelativeDistinguishedName>
        <RelativeDistinguishedName>
            <Field>organizationName</Field>
            <DirectoryServicesAttribute>company</DirectoryServicesAttribute>
            <Mandatory>true</Mandatory>
        </RelativeDistinguishedName>
        <RelativeDistinguishedName>
            <Field>organizationalUnitName</Field>
            <DirectoryServicesAttribute>department</DirectoryServicesAttribute>
            <Mandatory>true</Mandatory>
        </RelativeDistinguishedName>
        <RelativeDistinguishedName>
            <Field>localityName</Field>
            <DirectoryServicesAttribute>l</DirectoryServicesAttribute>
        </RelativeDistinguishedName>
        <RelativeDistinguishedName>
            <Field>countryName</Field>
            <DirectoryServicesAttribute>c</DirectoryServicesAttribute>
        </RelativeDistinguishedName>
    </SubjectDistinguishedName>
  </DirectoryServicesMapping>
</CertificateRequestPolicy>
```

The following attributes are set in the Computer-Account:
![grafik](https://user-images.githubusercontent.com/76774979/205027779-ae15035e-858d-4ded-8bf8-f0a1c7b1d99c.png)

However, due to the required attributes organizationName and organizationalUnitName, I get the following warning from TameMyCerts:
```
Request 65 for DC was denied because:
The directory attribute "company" that was specified for construction of the mandatory "organizationName" subject relative distinguished name is either not supported or not populated for DC01.Firma.local.
The directory attribute "department" that was specified for construction of the mandatory "organizationalUnitName" subject relative distinguished name is either not supported or not populated for DC01.Firma.local.
```

If I set Mandatory to False for the two values, the certificate is issued, without the two attributes but with other attributes of the AD Computer-Account:
![grafik](https://user-images.githubusercontent.com/76774979/205028749-26a04952-bb9c-4f1d-a77f-e68b9b0b058c.png)

So he doesn't seem to be able to find certain attributes, although they are in the account (can also be read with LDAP).
AD attribute "info" doesn't work either.
Testing using AD attribute "l" (also used for localityName) on organizationName works, but that's not the goal. ;)

Where is the mistake in my way of thinking?

**Comment by Sleepw4lker** [comment]
Hello, thanks for reaching out. Unfortunately, I cannot reproduce the issue (works fine in my lab).

It seems you did everything right, so I'd be eager to have a look into your lab if it's ok for you.

If you like, reach out to me so that we can have a look into this together, e.g. via screen sharing.

**Comment by Sleepw4lker** [comment]
Hi @Thommii as we found out the Global Catalog was not available, thus attributes not present in the GC were not available. Root cause is that TameMyCerts queries the GC when SearchRoot is not specified in the policy file.

**Comment by Sleepw4lker** [comment]
Fix has been implemented and will be available soon with v1.4 release.
